File: Code/C#/HVPSAPI/Messages/ErrorMessage.cs
using Core.Messages.Messages;
using HVPSAPI.DataMemberNames.Messages;
using HVPSCore.Enums;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace HVPSAPI.Messages
{
    [DataContract]
    public class ErrorMessage : TypedMessageBase
    {

        [JsonPropertyName(ErrorMessageDataMemberNames.ErrorType)]
        [JsonInclude]
        [DataMember(Name = ErrorMessageDataMemberNames.ErrorType)]
        public ErrorType ErrorType { get; protected set; }

        [JsonPropertyName(ErrorMessageDataMemberNames.SerializedError)]
        [JsonInclude]
        [DataMember(Name = ErrorMessageDataMemberNames.SerializedError)]
        public string SerializedError { get; protected set; }
        public ErrorMessage()
            : base()
        {
            Type = MessageTypes.Error;
        }
    }
}

File: Code/JavaScript/client/src/messages/ErrorMessage.js
export default class ErrorMessage
{
 static toJSON(o){ 
    const r = {};
   r["tpe"]="e";
    r["t"]=o.errorType;
    r["s"]=o.serializedError;
    r["r"]=o.reason;
    return r;
 }
 static fromJSON(o){
    const r = {};
    r.errorType=o["t"];
    r.serializedError=o["s"];
    r.reason=o["r"];
   return r;
 }
}